cypress/integration/cluster_spec.js

```diff
@@ -0,0 +1,11 @@
+describe("Cluster functionality test", function () {
+  it("Add cluster button opens modal", function () {
+    cy.visit("localhost:9000");
+    cy.get("#add-cluster-btn").then(() => {
+      cy.get("#add-cluster-btn")
+        .first()
+        .click();
+    });
+    cy.contains('Add Cluster')
```

Comment:
Can you build this out further to run through a full cluster add?

Reply:
Added tests for cluster adding with valid data